pkg/rtc/participant.go

```diff
@@ -666,50 +668,84 @@ func (p *ParticipantImpl) GetAudioLevel() (level uint8, active bool) {
 	return
 }
 
-func (p *ParticipantImpl) GetConnectionQuality() livekit.ConnectionQuality {
+func (p *ParticipantImpl) GetConnectionQuality() *livekit.ConnectionQualityInfo {
 	// avg loss across all tracks, weigh published the same as subscribed
 	var pubLoss, subLoss uint32
 	var reducedQualityPub bool
 	var reducedQualitySub bool
+	var audioScores float64
+	var numAudioTracks int
 	p.lock.RLock()
 	defer p.lock.RUnlock()
 	for _, pubTrack := range p.publishedTracks {
 		if pubTrack.IsMuted() {
 			continue
 		}
-		pubLoss += pubTrack.PublishLossPercentage()
-		publishing, registered := pubTrack.NumUpTracks()
-		if registered > 0 && publishing != registered {
-			reducedQualityPub = true
+		// audio has scores available
+		if pubTrack.Kind() == livekit.TrackType_AUDIO {
+			audioScores += pubTrack.GetConnectionScore()
```

Comment:
currently I think it's a bit confusing where `GetConnectionScore` only returns valid values for audio, where as for video, it doesn't return anything useful, and we have to compute it here.

If we are surfacing a `GetConnectionScore` on the published and subscribed track, why don't we move the video logic in there too?